src/lib/optimizer/strategy/projection_combination_rule.cpp

```diff
@@ -0,0 +1,92 @@
+#include "projection_combination_rule.hpp"
+
+#include <memory>
+#include <string>
+#include <vector>
+
+#include "logical_query_plan/abstract_lqp_node.hpp"
+#include "logical_query_plan/projection_node.hpp"
+
+namespace opossum {
+
+std::string ProjectionCombinationRule::name() const { return "Projection Combination Rule"; }
+
+bool ProjectionCombinationRule::apply_to(const std::shared_ptr<AbstractLQPNode>& node) {
```

Comment:
Same as for the ConstantCalculationRule: You are removing nodes that "produce" columns, so you somehow need to fix all the Nodes further up in the tree that reference to these nodes.
ConstantCalculationRule and ProjectionCombinationRule might be able to share some code here.